packages/block-library/src/navigation/index.php

```diff
@@ -253,8 +253,15 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 
 	$inner_blocks = $block->inner_blocks;
 
-	// If `__unstableLocation` is defined, create inner blocks from the classic menu assigned to that location.
-	if ( empty( $inner_blocks ) && array_key_exists( '__unstableLocation', $attributes ) ) {
+	// If `__unstableLocation` is defined and:
+	// - we have menu items at the defined location
+	// - we don't have a relationship to a `wp_navigation` Post (via `navigationMenuId`).
+	// ...then create inner blocks from the classic menu assigned to that location.
+	if (
+		array_key_exists( '__unstableLocation', $attributes ) &&
+		! array_key_exists( 'navigationMenuId', $attributes ) &&
+		! empty( gutenberg_get_menu_items_at_location( $attributes['__unstableLocation'] ) )
+	) {
 		$menu_items = gutenberg_get_menu_items_at_location( $attributes['__unstableLocation'] );
```

Comment:
There is now an inefficiency here as we are calling `gutenberg_get_menu_items_at_location` twice...

Reply:
Let's follow up here.